lib/services/local/saml_idp_service_provider.go

```diff
@@ -198,6 +202,9 @@ func (s *SAMLIdPServiceProviderService) UpdateSAMLIdPServiceProvider(ctx context
 
 // DeleteSAMLIdPServiceProvider removes the specified SAML IdP service provider resource.
 func (s *SAMLIdPServiceProviderService) DeleteSAMLIdPServiceProvider(ctx context.Context, name string) error {
+	if err := spReferencedByAWSICPlugin(ctx, s.backend, name); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
The deletion prevention mechanism used for integrations https://github.com/gravitational/teleport/pull/48052 uses AtomicWrite but here I have not used it. Mainly because the service provider itself is created by the plugin. As such, there is no situation where a plugin exist and does not reference the service provider and the service provider is always created before the plugin is created.

Open to suggestion otherwise.